src/main/java/org/folio/rest/impl/BLUsersAPI.java

```diff
@@ -914,4 +931,163 @@ else if(include.get(i).equals(SERVICEPOINTS_INCLUDE)) {
+              .map(o -> ((JsonObject) o).getString("value"))
+              .flatMap(s -> Stream.of(s.split("[^\\w\\.]+")))
+              .collect(Collectors.toList()));
+        })
+      );
+    } catch (UnsupportedEncodingException e) {
+      future.fail(e);
+    }
+    return future;
+  }
+
+  /**
+   *
+   * @param userToNotify
+   * @return
+   */
+  private io.vertx.core.Future<Void> sendResetPasswordNotification(User userToNotify) {
+    //TODO: should be implemented once notification functionality is completed.
+    return io.vertx.core.Future.succeededFuture();
+  }
+
+  /**
+   *
+   * @param fieldAliasList list of aliases to use [LOCATE_USER_USERNAME LOCATE_USER_PHONE_NUMBER LOCATE_USER_EMAIL]
+   * @param entity - an identity with a value
+   * @param okapiHeaders
+   * @return
+   */
+  private io.vertx.core.Future<Void> doPostBlUsersForgotten(List<String> fieldAliasList, Identifier entity,
+                                                            java.util.Map<String, String> okapiHeaders) {
+    io.vertx.core.Future<Void> asyncResult = io.vertx.core.Future.future();
+    getLocateUserFields(fieldAliasList, okapiHeaders).setHandler(locateUserFieldsAR -> {
+      if (locateUserFieldsAR.succeeded()) {
+        String query = buildQuery(locateUserFieldsAR.result(), entity.getId());
+
+        String tenant = okapiHeaders.get(OKAPI_TENANT_HEADER);
+        String okapiURL = okapiHeaders.get(OKAPI_URL_HEADER);
+
+        HttpClientInterface client = HttpClientFactory.getHttpClient(okapiURL, tenant);
+
+        okapiHeaders.remove(OKAPI_URL_HEADER);
+
+        try {
+          StringBuilder userUrl = new StringBuilder("/users?").append("query=").append(URLEncoder.encode(query, "UTF-8")).append("&").append("offset=0&limit=2");
+
+          client.request(userUrl.toString(), okapiHeaders).thenAccept(userResponse -> {
+            String noUserFoundMessage = "User is not found: ";
+
+            if(!responseOk(userResponse)) {
+              asyncResult.fail(new NoSuchElementException(noUserFoundMessage + entity.getId()));
+            } else {
+              JsonArray users = userResponse.getBody().getJsonArray("users");
+              int arraySize = users.size();
+              if (arraySize == 0 || arraySize > 1) {
+                asyncResult.fail(new NoSuchElementException(noUserFoundMessage + entity.getId()));
+              } else {
+                try {
+                  User user = (User) Response.convertToPojo(users.getJsonObject(0), User.class);
+                  sendResetPasswordNotification(user).setHandler(asyncResult::handle);
+                } catch (Exception e) {
+                  asyncResult.fail(e);
+                }
+              }
+            }
+          });
+
+        } catch (Exception e) {
+          asyncResult.fail(e);
+        }
+
+      } else {
+        asyncResult.fail(locateUserFieldsAR.cause());
+      }
+
+    });
+
+    return asyncResult;
+  }
```

Comment:
You can reduce indentation and else clauses by inserting `return;` after `asyncResult.fail(...)`. For example
```
if (locateUserFieldsAR.failed()) {
  asyncResult.fail(locateUserFieldsAR.cause());
  return;
}
```
```
if(!responseOk(userResponse)) {
  asyncResult.fail(new NoSuchElementException(noUserFoundMessage + entity.getId()));
  return;
}
```

Reply:
Done